src/sagemaker_huggingface_inference_toolkit/transformers_utils.py

```diff
@@ -202,7 +207,9 @@ def _load_model_from_hub(
     # check if safetensors weights are available
     if framework == "pytorch":
         files = HfApi().model_info(model_id).siblings
-        if any(f.rfilename.endswith("safetensors") for f in files):
+        if is_optimum_neuron_available() and any(f.rfilename.endswith("neuron") for f in files):
+            framework = "neuronx"
```

Comment:
In optimum, both models compiled for inf1 and inf2 are in `.neuron` format. If we want to distinguish that, we could leverage the compiler type under `config.neuron`: `neuronx-cc` for neuronx and `neuron-cc` for neuron.

Reply:
thats fine since inf1 should be supported for encoder tasks as well.